### PR TITLE
Collectors: Warn unfulfilled aspects

### PR DESCRIPTION
**Pull request by adhikasp**
Show warning if some of requested aspects could not be fulfilled
by coala.

Closes https://github.com/coala/coala/issues/4656

### CODE REVIEW

File: coalib/collecting/Collectors.py [hunk 1]
```diff
@@ -220,16 +221,24 @@ def collect_bears_by_aspects(aspects, kinds):
...
             if (aspect in bear.aspects['detect'] or
                     aspect in bear.aspects['fix']):
+                found = True
                 index = kinds.index(_get_kind(bear))
                 # Avoid duplicate
                 if bear not in bears_found[index]:
                     bears_found[index].append(bear)
                 break
+        if not found:
+            unfulfilled_aspects.append(type(aspect).__qualname__)
```
Comment (jayvdb) [inline coalib/collecting/Collectors.py]: Use for..else

File: tests/collecting/CollectorsTest.py [hunk 2]
```diff
@@ -288,6 +292,25 @@ def test_aspect_bear(self):
...
+            local_bears, global_bears = collect_bears_by_aspects(
+                aspects,
+                [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL])
+        self.assertRegex(log.output[0],
+                         'coala cannot find bear that could analyze the '
+                         'following aspects: \['
+                         "'Root\.Redundancy\.UnusedVariable\.UnusedParameter'"
+                         '\]')
+
+        self.assertEqual(len(global_bears), 0)
```
Comment (jayvdb) [inline tests/collecting/CollectorsTest.py]: AssertEqual to empty list. Avoid len below also.

File: tests/collecting/CollectorsTest.py [hunk 3]
```diff
@@ -288,6 +292,25 @@ def test_aspect_bear(self):
         self.assertEqual(len(local_bears), 1)
         self.assertIs(local_bears[0].name, 'AspectTestBear')
 
+    def test_collect_bears_by_aspects_unfulfilled_aspect(self):
```
Comment (jayvdb) [inline tests/collecting/CollectorsTest.py]: 'By-apects' is redundant

File: coalib/collecting/Collectors.py (hunk 1 above)
Comment (adhikasp) [inline coalib/collecting/Collectors.py]: Wow, I just learn about this!

**Comment by jayvdb** [comment]
ack 5001406

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**